iidm/iidm-impl/src/main/java/com/powsybl/iidm/network/impl/NetworkIndex.java

```diff
@@ -50,14 +53,54 @@ void checkAndAdd(Identifiable<?> obj) {
         all.add(obj);
     }
 
-    Identifiable get(String id) {
+    boolean addAlias(Identifiable<?> obj, String alias) {
+        if (objectsById.containsKey(alias)) {
+            if (objectsById.get(alias).equals(obj)) {
+                // Silently ignore affecting the objects id to its own aliases
+                return false;
+            }
+            Identifiable<?> aliasConflict = objectsById.get(alias);
```

Comment:
It's not efficient: you search for the object 3 times. If the map is huge, it could lead to bad performance issue.